node/src/mdns.rs

```diff
@@ -29,12 +29,11 @@ pub struct NmosMdnsRegistry {
 }
 
 impl NmosMdnsRegistry {
-    pub fn parse(discovery: &ServiceDiscovery) -> Option<Self> {
+    pub fn parse(discovery: &ServiceDiscovery, api_version: &APIVersion) -> Option<Self> {
```

Comment:
Could also be implemented as `impl TryFrom<(&ServiceDiscovery, &ApiVersion)> for  NmosMdnsRegistry`.
The term parse is mostly used in conjuction with FromStr/ building things from string types.

Reply:
Also using a Result here seems more appropiate to me.